runtime/vm/jvminit.c

```diff
@@ -6000,6 +6002,291 @@ processCompressionOptions(J9JavaVM *vm){
+					if (NULL == upToColon) {
+						/* `file=` with no filename specified */
+						goto xlogerr;
+					}
+					gclogfile = upToColon;
+				} else {
+					/* Anything else such as `<something>=<something>` is an error */
+					goto xlogerr;
+				}
+
+				if (NULL != fullOptions) {
+					/* Anything besides trailing ':'s is unrecognized */
+					if (('\0' != *fullOptions) && (':' != *fullOptions)) {
+						unrecognizedOption = TRUE;
+					} else {
+						strUpToDelimiter(&fullOptions, ':');
+						if ((NULL != fullOptions) && ('\0' != *fullOptions)) {
+							unrecognizedOption = TRUE;
+						}
+					}
+				}
+			}
+		}
+#if defined(DEBUG_XLOG)
+		j9tty_printf(PORTLIB, "setgclog %d gclog %d file %s\n", setgclog, gclog, gclogfile);
+#endif /* defined(DEBUG_XLOG) */
+		if (setgclog && gclog) {
+		 	if (NULL != gclogfile) {
+				IDATA mapRc = 0;
+				char *buf = NULL;
+				char timeBuf[20];
+				UDATA sizeRequired = 0;
+				struct J9StringTokens *stringTokens = j9str_create_tokens(j9time_current_time_millis());
+				if (NULL == stringTokens) {
+					rc = JNI_ERR;
+					goto xlogret;
+				}
+				j9str_subst_tokens(timeBuf, sizeof(timeBuf), "%Y-%m-%d_%H-%M-%S", stringTokens);
+
+				if (j9str_set_token(PORTLIB, stringTokens, "p", "%lld", j9sysinfo_get_pid())
+					|| j9str_set_token(PORTLIB, stringTokens, "t", "%s", timeBuf)
+				) {
+					j9str_free_tokens(stringTokens);
+					rc = JNI_ERR;
+					goto xlogret;
+				}
+				sizeRequired = j9str_subst_tokens(NULL, 0, gclogfile, stringTokens);
+				/* sizeRequired already includes space for the NULL */
+				buf = j9mem_allocate_memory(sizeof(MAPOPT_XVERBOSEGCLOG) - 1 + sizeRequired, OMRMEM_CATEGORY_VM);
+				if (NULL == buf) {
+					j9str_free_tokens(stringTokens);
+					rc = JNI_ERR;
+					goto xlogret;
+				}
+				strcpy(buf, MAPOPT_XVERBOSEGCLOG);
+				j9str_subst_tokens(buf + sizeof(MAPOPT_XVERBOSEGCLOG) - 1, sizeRequired, gclogfile, stringTokens);
+				j9str_free_tokens(stringTokens);
+				j9tty_printf(PORTLIB, "mapping %s to %s\n", j9vm_args->actualVMArgs->options[xlogindex].optionString, buf);
+				mapRc = registerCmdLineMapping(vm, j9vm_args->actualVMArgs->options[xlogindex].optionString, buf, EXACT_MAP_NO_OPTIONS);
+				j9mem_free_memory(buf);
+				if (RC_FAILED == mapRc) {
+					rc = JNI_ERR;
+					goto xlogret;
+				}
+			} else {
+				if (RC_FAILED == registerCmdLineMapping(vm, j9vm_args->actualVMArgs->options[xlogindex].optionString, MAPOPT_VERBOSE_GC, EXACT_MAP_NO_OPTIONS)) {
+					rc = JNI_ERR;
+					goto xlogret;
+				}
+			}
+		} else if (setgclog) {
+			/* ATM there is no OpenJ9 option to disable gc logging */
+			unrecognizedOption = TRUE;
+		} else {
+			unrecognizedOption = TRUE;
+		}
+		if (unrecognizedOption) {
+			CONSUME_ARG(j9vm_args, xlogindex);
+			j9nls_printf(PORTLIB, J9NLS_WARNING, J9NLS_VM_UNRECOGNISED_CMD_LINE_OPT, j9vm_args->actualVMArgs->options[xlogindex].optionString);
+		}
+
+xlogret:
+		if (xlogoptions != xlogoptionsbuf) {
```

Comment:
The l-value (constant) `xlogoptionsbuf` should be on the left.

Reply:
Fixed